src/db/file/repo.js

```diff
@@ -34,6 +34,22 @@ exports.getRepo = async (name) => {
   });
 };
 
+exports.getRepoByUrl = async (url) => {
```

Comment:
Keep me honest, but can we not enforce uniqueness with `organization`, `name` and `provider`, i.e. `zsh-org`, `zsh` and `gitlab`.

Can we take advantage of the fact that both GitHub and GitLab follow the same naming convention for repositories and enforce uniqueness this way?

GitHub: `finos`, `git-proxy`, `github`
GitLab: `zsh-org`, `zsh`, `gitlab`